docs/app/js/app.js

```diff
@@ -881,4 +883,59 @@ function($rootScope, $scope, component, demos, $templateRequest) {
       }
     }
   };
-});
+})
+
+.factory('scrollCache', function() {
+  var cache = {};
+  var linkClicked = false;
+
+  function setScroll(key, scrollPostion) {
+    cache[key] = scrollPostion;
+  }
+
+  function getScroll(key) {
+    return cache[key] || 0;
+  }
+
+  return {
+    getScroll: getScroll,
+    setScroll: setScroll,
+    linkClicked: linkClicked
+  };
+})
+/** This directive caches the scroll position for each route + templateUrl combination
+ * This helps in restoring the scroll when the user visits the same page again
```

Comment:
Let's make this a little more accurate.

"This helps in restoring the scroll when the user uses back or forward navigation to return to a page."